api/src/main/java/org/pmiops/workbench/db/dao/UserService.java

```diff
@@ -80,11 +84,23 @@ private User updateWithRetries(Function<User, User> userModifier, User user) {
 
   private void updateDataAccessLevel(User user) {
     if (user.getDataAccessLevel() == DataAccessLevel.UNREGISTERED) {
+      String userEmailVerificationStatus = null;
+      if(user.getContactEmail() != null) {
+        try {
+          userEmailVerificationStatus = mailChimpService.getMember(user.getContactEmail());
```

Comment:
See above, wondering if we can cache VERIFIED on User to avoid repeating this call.

Reply:
We only need to do this if the status is pending...

We may want to have a method on MailChimpService that is like getVerificationStatus(User)... it can handle the logic of mapping our enums and MailChimpService status to an e-mail verification status, and updating User if appropriate..